lib/identity_cache.rb

```diff
@@ -240,6 +242,24 @@ def with_deferred_parent_expiration
       Thread.current[:idc_parent_records_for_cache_expiry].clear
     end
 
+    def with_deferred_attribute_expiration
```

Comment:
I agree it would be nice to collapse those two.

Maybe `with_deferred_expiration` that calls both `with_deferred_parent_expiration` and `with_deferred_attribute_expiration`.

Or `with_deferred_expiration(parent: true, attribute: false` as a way to control that.

Reply:
Agree. I am going to merge the two methods, and test again. 👍